owslib/csw.py

```diff
@@ -658,7 +660,7 @@ def _invoke(self):
 
         if isinstance(self.request, six.string_types):  # GET KVP
             self.request = '%s%s' % (bind_url(request_url), self.request)
-            self.response = openURL(self.request, None, 'Get', username=self.username, password=self.password, timeout=self.timeout).read()
+            self.response = self.auth.openURL(self.request, None, 'Get', timeout=self.timeout).read()
```

Comment:
Should `openURL` be a function of the new Authentication mechanism or should we do it the other way around (i.e. pass an auth object as a parameter to `openURL`)

Reply:
Its up to you. I was thinking make the auth passing transparent, but can do it either way.

Reply:
@cehbrecht thoughts?

Reply:
Implemented